### ISSUE TITLE
Subclass demo code in eta-tutorials.rst won't compile

### ISSUE THREAD

**Issue by andrewgmcgregor** [typelead/eta#636]
I can’t compile this code copied from the project documentation:

```
$ cat SubclassExample.hs 
{-# LANGUAGE TypeFamilies, DataKinds, FlexibleContexts #-}

foreign import java unsafe toString :: (a <: Object) => a -> String

data File = File @java.io.File
  deriving Class

type instance Inherits File = '[Object]

main :: IO ()
main = do
  file <- java $ newFile "test.txt"
  -- This line will now typecheck!
  putStrLn (toString file)
```

```
$ ../../etlas/etlas/.stack-work/dist/x86_64-osx/Cabal-1.22.5.0/build/etlas/etlas exec eta -- SubclassExample.hs 
[1 of 1] Compiling Main             ( SubclassExample.hs, SubclassExample.jar )

SubclassExample.hs:3:43:
    Illegal operator ‘<:’ in type ‘a <: Object’
      Use TypeOperators to allow operators in types

SubclassExample.hs:3:46:
    Not in scope: type constructor or class ‘Object’

SubclassExample.hs:6:12:
    Not in scope: type constructor or class ‘Class’

SubclassExample.hs:8:15:
    Not in scope: type constructor or class ‘Inherits’

SubclassExample.hs:8:33:
    Not in scope: type constructor or class ‘Object’

SubclassExample.hs:12:11: Not in scope: ‘java’

SubclassExample.hs:12:18: Not in scope: ‘newFile’
```

I tried on the binary version of atlas I got from from eta-lang.org
```
$ ./etlas --version
etlas version 1.1.0.0
compiled using version 2.1.0.0 of the etlas-cabal library 
```

and I compiled branch 0.7.0b1
```
$ ../../etlas/etlas/.stack-work/dist/x86_64-osx/Cabal-1.22.5.0/build/etlas/etlas --version
etlas version 1.1.0.0
compiled using version 2.1.0.0 of the etlas-cabal library 
```

I am using macOS high sierra 10.13.2, but I also downloaded the binary and tried it on Linux and had the same problem

**Comment by andrewgmcgregor** [comment]
Got it working like by changing it to this:

```
{-# LANGUAGE TypeFamilies, DataKinds, FlexibleContexts, TypeOperators #-}

import Java hiding (toString)

foreign import java unsafe toString :: (a <: Object) => a -> String

data File = File @java.io.File
  deriving Class

foreign import java unsafe "@new" newFile  :: String -> Java a File

type instance Inherits File = '[Object]

main :: IO ()
main = do
  file <- java $ newFile "test.txt"
  -- This line will now typecheck!
  putStrLn (toString file)
```

```
$ ./etlas exec eta -- SubclassExample.hs  && java -jar RunSubclassExample.jar 
[1 of 1] Compiling Main             ( SubclassExample.hs, SubclassExample.jar )
Linking ./RunSubclassExample.jar ...
test.txt
```

Should I submit a pull request to update the documentation?

**Comment by rahulmutt** [comment]
@andrewgmcgregor Yes please!

**Comment by rahulmutt** [comment]
This should be resolved with #638.
